apps/clientside-bot/test/functions.test.ts

```diff
@@ -0,0 +1,1070 @@
+	createFAGCCategory,
+	createFAGCCommunity,
+	createFAGCReport,
+	createFAGCRevocation,
+	createGuildConfig,
+	createPrivateban,
+	createTimes,
+	createWhitelist,
+	randomElementsFromArray,
+	reportIntoFAGCBan,
+} from "./utils"
+import { Category, Community, Report } from "fagc-api-types"
+import faker from "faker"
+
+describe("splitIntoGroups", () => {
+	it("Should split a large array two smaller ones of an equal size", () => {
+		const items = new Array(1000).fill(0)
+		const groups = splitIntoGroups(items, 500)
+		expect(groups.length).toBe(2)
+		expect(groups[0].length).toBe(500)
+		expect(groups[1].length).toBe(500)
+	})
+	it("Should split a large array into multiple smaller ones if a smaller maxSize is provided", () => {
+		const items = new Array(1000).fill(0)
+		const groups = splitIntoGroups(items, 100)
+		expect(groups.length).toBe(10)
+		for (const group of groups) {
+			expect(group.length).toBe(100)
+		}
+	})
+	it("Should keep the original values of the inital array", () => {
+		const items = Array.from({ length: 1000 }, () =>
+			Math.floor(Math.random() * 10)
+		)
+		const groups = splitIntoGroups(items, 100)
+		expect(groups.flat()).toEqual(items)
+	})
+})
+
+describe("guildConfigChangedBanlists", () => {
+	let database: Connection
+	let categories: Category[]
+	let categoryIds: string[]
+	let communities: Community[]
+	let communityIds: string[]
+	beforeEach(async () => {
+		database = await createConnection({
+			type: "better-sqlite3",
+			database: ":memory:",
+			entities: [
+				FAGCBan,
+				InfoChannel,
+				BotConfig,
+				PrivateBan,
+				Whitelist,
+				Command,
+			],
+			synchronize: true,
+		})
+		categories = createTimes(createFAGCCategory, 100)
+		categoryIds = categories.map((x) => x.id)
+		communities = createTimes(createFAGCCommunity, 100)
+		communityIds = communities.map((x) => x.id)
+	})
+	afterEach(async () => {
+		await database.close()
+	})
+	it("Should create bans for reports that have just been included in the filters", async () => {
+		const oldGuildConfig = createGuildConfig({
+			categoryIds: [],
+			communityIds: [],
+		})
+		const newGuildConfig: typeof oldGuildConfig = {
+			...oldGuildConfig,
+			categoryFilters: categoryIds,
+			trustedCommunities: communityIds,
+		}
+
+		const reports = createTimes(
+			createFAGCReport,
+			[
+				{
+					categoryIds: newGuildConfig.categoryFilters,
+					communityIds: newGuildConfig.trustedCommunities,
+				},
+			],
+			5000
+		)
+
+		const results = await guildConfigChangedBanlists({
+			newConfig: newGuildConfig,
+			validReports: reports,
+			database,
+			allGuildConfigs: [newGuildConfig],
+		})
+
+		const fetchedFAGCBans = await database.getRepository(FAGCBan).find()
+
+		const allReportPlayernames = [
+			...new Set(reports.map((report) => report.playername)),
+		]
+		// the function should ban everyone there is a report against
+		expect(results.toBan.length).toBe(allReportPlayernames.length)
+		expect(results.toBan).toEqual(allReportPlayernames)
```

Comment:
The first assertion is not required because it cant be equal if the lengths are different.

Reply:
Yes, it is not, however, it is much easier to debug, as otherwise jest prints the whole difference of the two arrays, which can get pretty big if the method is giving wrong results.